### ISSUE TITLE
Migrating from Trac

### ISSUE THREAD

**Issue by bobemoe** [go-gitea/gitea#6908]
I'm looking to migrate my issues+wiki from Trac to Gitea and was wondering the best way to go about it?

I have found https://github.com/tracboat/tracboat but it is for migrating to GitLab, do you think it would work or be easily adaptable for use with Gitea? Or has anyone got a better idea?

Thanks :)

**Comment by bobemoe** [comment]
In the end I used this one, https://github.com/trustmaster/trac2github as I'm familiar with PHP than Python. I think it was simpler too, it only needed a few tweaks of URLs and Parameters to make it work with Gitea.

I'm happy. Close if you like, or keep open as it may be a nice feature?

**Comment by lunny** [comment]
@bobemoe I think we can just keep it open if someone would like to send a PR to Gitea.

**Comment by 6543** [comment]
there is a trac to gogs tool: http://strk.kbt.io/projects/go/trac2gogs maby fork it and integrate it?

**Comment by lunny** [comment]
@6543 maybe @strk like to contribute it. :)

**Comment by strk** [comment]
No free cycles, and for our case DB interaction is ok so trac2gogs would do :)

**Comment by stevejefferson** [comment]
FYI I have recently had the task of migrating a number of Trac projects to Gitea, several of which included multi-page Trac wikis.

To automate this I have built a command line utility which performs a "fairly comprehensive" migration of an entire Trac project to Gitea, including the wiki. While the starting point for this was `trac2gogs`, this utility has grown to cover a greater scope and the two codebases are now completely diverged.

As with `trac2gogs`, access to Gitea is via the database however all of the Gitea accesses (including those required by the wiki migration) have been abstracted out so hopefully this might prove a useful starting point should anyone want to produce a more integrated Trac migration in the future.

Having now migrated all of my Trac projects I have little remaining use for the utility myself however have uploaded the code to https://github.com/stevejefferson/trac2gitea in case it proves useful.

**Comment by 6543** [comment]
Yes ... there is  format similar to F3 that is internaly used for ex/im-port repos so you just would have to cover that format

**Comment by haumacher** [comment]
@6543 "format similar to F3" can you please give some more detail/pointer?

**Comment by 6543** [comment]
https://github.com/go-gitea/gitea/blob/main/services/migrations/restore.go
[exampleRepo.tar.gz](https://github.com/go-gitea/gitea/files/11132925/exampleRepo.tar.gz)
